pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ComparableOneOf.java

```diff
@@ -22,6 +22,7 @@ public record ComparableOneOf<E extends Enum<E>>(E kind, Comparable value) imple
         if (kind == null) {
             throw new NullPointerException("An enum 'kind' must be supplied");
         }
+        assert kind instanceof EnumWithProtoMetadata : "OneOf 'kind' must implement EnumWithProtoMetadata";
```

Comment:
Is an `assert` the right approach? Why not a regular check?

Reply:
Yes, it should suffice. The `hashCode()` implementation will error out with a class cast exception if this assertion is not satisfied. So as far as the correctness is concerned, the current implementation is solid.

Using an `assert` here helps improve the performance a little bit because we really don't expect to see anything but `EnumWithProtoMetadata` here. But like mentioned above, even if that happens, we'll error out later on then.